lib/spack/spack/report.py

```diff
@@ -247,30 +260,196 @@ class collect_info(object):
     Raises:
         ValueError: when ``format_name`` is not in ``valid_formats``
     """
-    def __init__(self, specs, format_name, filename):
-        self.specs = specs
+    def __init__(self, format_name, install_command):
         self.format_name = format_name
+        # Consider setting these properties in a more CDash specific place.
+        self.install_command = install_command
+        self.hostname = socket.gethostname()
+        self.osname = platform.system()
+        self.starttime = int(time.time())
+        self.buildstamp = time.strftime("%Y%m%d-%H%M-Experimental",
+                                        time.localtime(self.starttime))
 
         # Check that the format is valid
         if format_name not in itertools.chain(valid_formats, [None]):
             raise ValueError('invalid report type: {0}'.format(format_name))
 
-        self.filename = filename
-        self.collector = InfoCollector(specs) if self.format_name else None
-
     def __enter__(self):
         if self.format_name:
             # Start the collector and patch PackageBase.do_install
+            self.collector = InfoCollector(self.specs)
             self.collector.__enter__()
 
+    def cdash_initialize_report(self, report_data):
+        if not os.path.exists(self.filename):
+            os.mkdir(self.filename)
+        report_data['install_command'] = self.install_command
+        report_data['buildstamp'] = self.buildstamp
+        report_data['hostname'] = self.hostname
+        report_data['osname'] = self.osname
+
+    def cdash_build_report(self, report_data):
+        self.cdash_initialize_report(report_data)
+
+        # Mapping Spack phases to the corresponding CTest/CDash phase.
+        map_phases_to_cdash = {
+            'autoreconf': 'configure',
+            'configure':  'configure',
+            'build':      'build',
+            'install':    'build'
+        }
+
+        # Initialize data structures common to each phase's report.
+        cdash_phases = set(map_phases_to_cdash.values())
+        for phase in cdash_phases:
+            report_data[phase] = {}
+            report_data[phase]['log'] = ""
+            report_data[phase]['status'] = 0
+            report_data[phase]['starttime'] = self.starttime
+            report_data[phase]['endtime'] = self.starttime
+
+        # Track the phases we perform so we know what reports to create.
+        phases_encountered = []
+
+        # Parse output phase-by-phase.
+        regexp = re.compile(r"Executing phase: '(.*)'")
```

Comment:
more descriptive name for this regex might be nice, and if we're going to find phases in the output based on output from `PackageBase.do_install`, this should probably pull the text it's looking for from some variable in `PackageBase`, rather than rewriting here as text.